src/core/ext/filters/client_channel/retry_filter.cc

```diff
@@ -548,8 +551,9 @@ class RetryFilter::CallData {
   // If server_pushback is nullopt, retry_backoff_ is used.
   void StartRetryTimer(absl::optional<Duration> server_pushback);
 
-  static void OnRetryTimer(void* arg, grpc_error_handle error);
-  static void OnRetryTimerLocked(void* arg, grpc_error_handle error);
+  void OnRetryTimer();
+  static void OnRetryTimerLocked(void* arg,
+                                 GRPC_UNUSED grpc_error_handle error);
```

Comment:
`GRPC_UNUSED` isn't portable, right? I suspect that MSVC builds might complain if unused variable warnings were enabled.

https://github.com/grpc/grpc/blob/43c3d1f932da02c334e71dc5a9cd998755be5ff7/include/grpc/support/port_platform.h#L589-L595

Reply:
Thanks for catching it! `GRPC_UNUSED` is not portable and I couldn't find an equivalent for MSVC easily. There is the `[[maybe_unused]]` C++ attribute but it needs C++17.

So I went with this for now (based on our discussion):
```
GPR_DEBUG_ASSERT(error.ok());
#ifdef NDEBUG
  (void)error;
#endif
```

Reply:
The way we normally deal with this is to just comment out the parameter name, like this:

```
static void OnRetryTimerLocked(void* arg, grpc_error_handle /*error*/);
```

Reply:
In this case the argument is used only in debug build with `GPR_DEBUG_ASSERT`. Actually I should remove `GRPC_UNUSED` now since it is used in both debug and release build, just differently.

Reply:
Alternatively, consider removing the `GPR_DEBUG_ASSERT()`.  I'm not sure it's actually adding any value, since we're only ever scheduling this closure with OK status.

Reply:
Got it, done! Thanks for the suggestion!